core/controllers/contributor_dashboard.py

```diff
@@ -310,107 +310,91 @@ class ReviewableOpportunitiesHandler(
-        # that have in review translation suggestions.
+        # 1. Fetch the IDs of all published explorations in the topics'
+        #    published stories.
+        # 2. Fetch all suggestions that the user can review, and collect their
+        #    exploration IDs. Filter these IDs to only include published
+        #    explorations in published stories (see step 1).
+        # 3. Fetch all exploration opportunity summaries for the resulting set
+        #    of explorations.
+        # 4. Moved any pinned summaries to the top.
+
+        pinned_opportunity_summary = None
         if topic_name is None:
-            topics = topic_fetchers.get_all_topics()
+            topic_exp_ids = (
+                topic_services.get_all_published_story_exploration_ids())
         else:
             topic = topic_fetchers.get_topic_by_name(topic_name)
             if topic is None:
                 raise self.InvalidInputException(
                     'The supplied input topic: %s is not valid' % topic_name)
-            topics = [topic]
-        topic_stories = story_fetchers.get_stories_by_ids(
-            [
-                reference.story_id
-                for topic in topics
-                for reference in topic.get_all_story_references()
-                if reference.story_is_published
-            ],
-            strict=False
-        )
-        topic_exp_ids = []
-        for story in topic_stories:
-            if story is None:
-                continue
-            for node in story.story_contents.get_ordered_nodes():
-                if node.exploration_id is None:
-                    raise Exception(
-                        'No exploration_id found for the node_id: %s'
-                        % node.id
-                    )
-                topic_exp_ids.append(node.exploration_id)
+            topic_exp_ids = (
+                topic_services.get_all_published_story_exploration_ids(
+                    topic_id=topic.id))
+            if language and self.user_id:
+                pinned_opportunity_summary = (
+                    opportunity_services.get_pinned_lesson(
+                        self.user_id,
+                        language,
+                        topic.id
+                    ))
+
+        # TODO (#19664): Implement fetching target IDs using GAE projection
+        # queries.
         in_review_suggestions, _ = (
             suggestion_services
             .get_reviewable_translation_suggestions_by_offset(
                 user_id, topic_exp_ids, None, 0, None, language))
-        # Filter out suggestions that should not be shown to the user.
         # This is defined as a set as we only care about the unique IDs.
         in_review_suggestion_target_ids = {
-            suggestion.target_id
-            for suggestion in
-            suggestion_services.get_suggestions_with_editable_explorations(
-                in_review_suggestions)
```

Comment:
Why did we drop this condition? I suppose it is somewhat moot now -- we had an issue in the past where we couldn't actually edit explorations because of memory overflow errors, so we added this to avoid edits happening, but that issue is fixed now. That said, if this functionality still exists we should probably respect it here? (The alternative is to delete that functionality entirely but that will need more analysis.)

Reply:
Oh wow, that was 7 months ago :)

Sorry ignore my latest comment, I think what you have is fine. I'll try and file an issue to get the corresponding functionality dropped since I don't think we need it any more.